libs/markdown-to-html/src/MarkdownToHtml.tsx

```diff
@@ -53,15 +53,32 @@ const markdownToHtmlMap = {
   },
 }
 
-const disallowedElements = ['blockquote', 'code', 'h1', 'h5', 'h6', 'hr', 'img']
+const descriptionMarkdownToHtmlMap = {
+  p: ({ children }: PropsWithChildren) => <Paragraph size="small">{children}</Paragraph>,
+}
+
+const richTextAllowedElements = ['a', 'br', 'em', 'h2', 'h3', 'h4', 'li', 'ol', 'p', 'strong', 'ul']
+// Some screen readers do not read rich text content in input descriptions.
+// For this reason, we only allow paragraphs.
+const descriptionAllowedElements = ['p']
```

Comment:
Did you discuss this with Linda? Since it'll impact users without screen readers quite a lot.